pkg/kinflate/commands/util.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package commands
+
+import (
+	"errors"
+	"fmt"
+	"path"
+	"strings"
+
+	"github.com/ghodss/yaml"
+
+	manifest "k8s.io/kubectl/pkg/apis/manifest/v1alpha1"
+	"k8s.io/kubectl/pkg/kinflate/constants"
+	"k8s.io/kubectl/pkg/kinflate/util/fs"
+)
+
+func validateManifestFile(mPath string, fsys fs.FileSystem) (string, error) {
+	f, err := fsys.Stat(mPath)
+	if err != nil {
+		errorMsg := fmt.Sprintf("Manifest (%s) missing\nRun `kinflate init` first", mPath)
+		return "", errors.New(errorMsg)
+	}
+	if f.IsDir() {
+		mPath = path.Join(mPath, constants.KubeManifestFileName)
+		_, err = fsys.Stat(mPath)
+		if err != nil {
```

Comment:
what if `mpath` is still a dir at this point? :D

Reply:
By the join statement with `constants.KubeManifestFileName`, mPath should be a file path.

Reply:
Well, unless you have a dir named `Kube-manifest.yaml`. Though unlikely, I agree ;-)